pkg/testmachinery/controller/watch/watch_test.go

```diff
@@ -109,117 +120,125 @@ var _ = Describe("Watch", func() {
 		Expect(err).ToNot(HaveOccurred())
 	})
 
-	It("watch should timeout after 2 seconds", func() {
-		ctx, cancel := context.WithCancel(context.Background())
-		defer cancel()
+	for _, opts := range options {
 
-		w, err := watch.New(log.NullLogger{}, restConfig, nil)
-		Expect(err).ToNot(HaveOccurred())
-		go func() {
-			err := w.Start(ctx.Done())
-			Expect(err).ToNot(HaveOccurred())
-		}()
+		Context(string(opts.InformerType), func() {
+			It("watch should timeout after 2 seconds", func() {
+				ctx, cancel := context.WithCancel(context.Background())
+				defer cancel()
 
-		err = watch.WaitForCacheSyncWithTimeout(w, 2*time.Minute)
-		Expect(err).ToNot(HaveOccurred())
+				w, err := watch.New(log.NullLogger{}, restConfig, opts)
+				Expect(err).ToNot(HaveOccurred())
+				go func() {
+					err := w.Start(ctx.Done())
+					Expect(err).ToNot(HaveOccurred())
+				}()
 
-		startTime := time.Now()
-		err = w.WatchUntil(2*time.Second, "test", "test", func(tr *tmv1beta1.Testrun) (bool, error) { return false, nil })
-		Expect(err).To(HaveOccurred())
-		Expect(time.Now().Sub(startTime).Seconds()).To(BeNumerically("~", 2, 0.01))
-	})
+				err = watch.WaitForCacheSyncWithTimeout(w, 2*time.Minute)
+				Expect(err).ToNot(HaveOccurred())
 
-	It("watch should reconcile once", func() {
-		ctx, cancel := context.WithCancel(context.Background())
-		defer cancel()
-		w, err := watch.New(log.NullLogger{}, restConfig, nil)
-		Expect(err).ToNot(HaveOccurred())
-		go func() {
-			err := w.Start(ctx.Done())
-			Expect(err).ToNot(HaveOccurred())
-		}()
+				startTime := time.Now()
+				err = w.WatchUntil(2*time.Second, "test", "test", func(tr *tmv1beta1.Testrun) (bool, error) { return false, nil })
+				Expect(err).To(HaveOccurred())
+				Expect(time.Now().Sub(startTime).Seconds()).To(BeNumerically("~", 2, 0.01))
```

Comment:
Let's see if that precision is not too strict and the test will flake (i.e. because of resource/cpu limits), so let's try :)